src/Storages/StorageMergeTree.cpp

```diff
@@ -297,6 +297,13 @@ void StorageMergeTree::checkTableCanBeDropped([[ maybe_unused ]] ContextPtr quer
         return;
 
     auto table_id = getStorageID();
+    const auto & storage_settings = getSettings();
+    if (storage_settings->max_table_size_to_drop >= 0)
+    {
+        getContext()->checkTableCanBeDropped(table_id.database_name, table_id.table_name, getTotalActiveSizeInBytes(), storage_settings->max_table_size_to_drop);
```

Comment:
Similar code in ReplicatedStorageMergeTree should be adjusted - https://github.com/ClickHouse/ClickHouse/blob/9b517e47129bd52f509cdb073b822f6b93f8f6cf/src/Storages/StorageReplicatedMergeTree.cpp#L6449